app/models/etmp/EtmpDisplayRegistration.scala

```diff
@@ -18,13 +18,44 @@ package models.etmp
 
 import play.api.libs.json.{Json, OFormat}
 
+import java.time.LocalDate
+
 case class EtmpDisplayRegistration(
                                     tradingNames: Seq[EtmpTradingName],
                                     schemeDetails: EtmpDisplaySchemeDetails,
                                     bankDetails: EtmpBankDetails,
                                     exclusions: Seq[EtmpExclusion],
                                     adminUse: EtmpAdminUse
-                                  )
+                                  ) {
+
+
+  def canRejoinRegistration(currentDate: LocalDate): Boolean = {
+    exclusions.lastOption match {
+      case Some(etmpExclusion) if etmpExclusion.exclusionReason == EtmpExclusionReason.Reversal => false
+      case Some(etmpExclusion) if isQuarantinedAndAfterTwoYears(currentDate, etmpExclusion) => true
+      case Some(etmpExclusion) if notQuarantinedAndAfterEffectiveDate(currentDate, etmpExclusion) => true
+      case _ => false
+    }
+  }
+
+  private def isQuarantinedAndAfterTwoYears(currentDate: LocalDate, etmpExclusion: EtmpExclusion): Boolean = {
+    if(etmpExclusion.quarantine) {
+      val minimumDate = currentDate.minusYears(2)
+      etmpExclusion.effectiveDate.isBefore(minimumDate)
+    } else {
+      false
+    }
+  }
+
+  private def notQuarantinedAndAfterEffectiveDate(currentDate: LocalDate, etmpExclusion: EtmpExclusion): Boolean = {
+    if(!etmpExclusion.quarantine) {
+      etmpExclusion.effectiveDate.isBefore(currentDate) || etmpExclusion.effectiveDate.isEqual(currentDate)
```

Comment:
Could add this file https://github.com/hmrc/ioss-exclusions-frontend/blob/0d195d3ada8cd70cf231238ca0bb33567c274ece/app/date/package.scala to the project and use `etmpExclusion.effectiveDate <= currentDate`

Reply:
I also changed this in the other front end. Joy of copy paste coding.

https://github.com/hmrc/ioss-returns-frontend/blob/181729a186cf397336f8004b9a2d81f667d5e605/app/models/etmp/EtmpDisplayRegistration.scala#L52